### PR TITLE
Fix DB initialization when missing database YAML file

### LINKED ISSUE #9811: Docker image from metasploitframework/metasploit-framework :  Unknown command: db_connect.
## Steps to reproduce

How'd you do it?

1. `wget https://github.com/rapid7/metasploit-framework/blob/master/docker-compose.yml`
2. `docker-compose run --rm --service-ports -e MSF_UID=$(id -u) -e MSF_GID=$(id -g) ms`

I followed the *README.md* located here : https://github.com/rapid7/metasploit-framework/tree/master/docker

## Expected behavior

What should happen?

msf_db should connect to postgre DB

## Current behavior

What happens instead?

The container can't found db_connect : 

```
$ sudo docker-compose run --rm --service-ports -e MSF_UID=$(id -u) -e MSF_GID=$(id -g) ms
Starting msf_db_1 ... done
[-] ***rting the Metasploit Framework console...-
[-] * WARNING: No database support: NilClass 
[-] ***
                                                  

 ______________________________________________________________________________
|                                                                              |
|                          3Kom SuperHack II Logon                             |
|______________________________________________________________________________|
|                                                                              |
|                                                                              |
|                                                                              |
|                 User Name:          [   security    ]                        |
|                                                                              |
|                 Password:           [               ]                        |
|                                                                              |
|                                                                              |
|                                                                              |
|                                   [ OK ]                                     |
|______________________________________________________________________________|
|                                                                              |
|                                                       https://metasploit.com |
|______________________________________________________________________________|


       =[ metasploit v5.0.0-dev                           ]
+ -- --=[ 1750 exploits - 1003 auxiliary - 304 post       ]
+ -- --=[ 536 payloads - 40 encoders - 10 nops            ]
+ -- --=[ ** This is Metasploit 5 development branch **   ]

[*] Processing docker/msfconsole.rc for ERB directives.
[*] resource (docker/msfconsole.rc)> Ruby Code (236 bytes)
LHOST => 172.22.0.3
[-] Unknown command: db_connect.
```

### Metasploit version

```
msf5 > version
Framework: 5.0.0-dev
Console  : 5.0.0-dev
```

### I installed Metasploit with:
- [ ] Kali package via apt
- [ ] Omnibus installer (nightly)
- [ ] Commercial/Community installer (from http://www.rapid7.com/products/metasploit/download.jsp)
- [ ] Source install (please specify ruby version)
- [X] Docker image from https://hub.docker.com/r/metasploitframework/metasploit-framework/

### OS

Alpine Linux (*See Dockerfile*)

### PR DESCRIPTION
**Pull request by mkienow-r7**
Fixes DB initialization when the database YAML file is missing. The message "WARNING: No database support: NilClass" reported during msfconsole startup is the first indicator of the issue. Subsequent DB operations fail to work as expected due to the framework not loading `CommandDispatcher::Db`. Fixes #9811

This also cleans up some things that were not caught in the early Goliath changes.

## Verification
- [x] Temporarily move `database.yml` file: `mv ~/.msf4/database.yml ~/.msf4/database.yml.disable`
- [x] Start `msfconsole`
- [x] Run command: `db_status`
- [x] **Verify** Command ran successfully and it reports `postgresql selected, no connection`
- [x] Connect to DB: `db_connect -y ~/.msf4/database.yml.disable`
- [x] **Verify** Command ran successfully and `db_status` reports it is connected to the specified database (`postgresql connected to ...`)
- [x] Verify ability to connect to a remote data store (Goliath)
   - [x] Start remote data service on a server or VM: `msfdb_ws`
   - [x] Connect to remote data service: `data_services --add <host>`
- [x] Restore `database.yml` file: `mv ~/.msf4/database.yml.disable ~/.msf4/database.yml`

### CODE REVIEW

**Comment by wvu-r7** [comment]
:clap:

File: lib/msf/core/db_manager.rb [hunk 1]
```diff
@@ -153,9 +157,6 @@ def check
   #
   def initialize_database_support
     begin
-      # Database drivers can reset our KCODE, do not let them
-      $KCODE = 'NONE' if RUBY_VERSION =~ /^1\.8\./
```
Comment (jbarnett-r7) [inline lib/msf/core/db_manager.rb]: Was this just removed because we no longer support Ruby 1.8?

Comment (wvu-r7) [reply lib/msf/core/db_manager.rb]: It's unfortunate `KCODE` doesn't mean Konami Code.

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 2]
```diff
@@ -1653,8 +1653,8 @@ def cmd_db_nmap_tabs(str, words)
   # Database management
   #
   def db_check_driver
-    if(not framework.db.driver)
-      print_error("No database driver installed. Try 'gem install pg'")
```
Comment (jbarnett-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: Why was the suggestion to install the pg gem removed here?

File: lib/msf/ui/console/command_dispatcher/db.rb [hunk 3]
```diff
@@ -1909,6 +1909,12 @@ def each_host_range_chunk(host_ranges, &block)
   #######
 
   def add_data_service(*args)
+    # database is required to use Mdm objects
+    unless framework.db.active
+      print_error("Database not connected; connect to an existing database with db_connect before using data_services")
```
Comment (jbarnett-r7) [inline lib/msf/ui/console/command_dispatcher/db.rb]: I feel like longer term we should move everything under a single command (either db_connect or data_services). Agree/disagree?

Comment (mkienow-r7) [reply lib/msf/ui/console/command_dispatcher/db.rb]: Agree

File: lib/msf/ui/console/command_dispatcher/db.rb (hunk 2 above)
Comment (mkienow-r7) [reply lib/msf/ui/console/command_dispatcher/db.rb]: I believe the suggestion to `gem install pg` is not relevant given we now use `bundler`. Is this not correct @busterb?

File: lib/msf/core/db_manager.rb (hunk 1 above)
Comment (mkienow-r7) [reply lib/msf/core/db_manager.rb]: @jbarnett-r7 Yes

Comment (mkienow-r7) [reply lib/msf/core/db_manager.rb]: @wvu-r7 :arrow_up::arrow_up::arrow_down::arrow_down::arrow_left::arrow_right::arrow_left::arrow_right::b::a:

**Comment by mkienow-r7** [comment]
# Release Notes

Fixes issues introduced with Goliath #9717 that caused confusing error messages and subsequent DB operations to fail to work as expected due to the framework not loading CommandDispatcher::Db when the database YAML file (`database.yml`) is missing.

**Pull request merged**